modules/core/shared/src/main/scala-3/io/circe/derivation/ConfiguredEnumDecoder.scala

```diff
@@ -0,0 +1,27 @@
+package io.circe.derivation
+
+import scala.deriving.Mirror
+import scala.compiletime.constValue
+import Predef.genericArrayOps
+import io.circe.{ Decoder, DecodingFailure, HCursor }
+
+trait ConfiguredEnumDecoder[A] extends Decoder[A]
```

Comment:
Looking at the code with fresh eyes I would say `DerivedInstance` could be deleted. It is mainly used in `ConfiguredDecoder` and `ConfiguredEncoder`, but  ConfiguredEncoder only uses the `elemLabels`. In both cases the core logic is in the `Configured*` creation, aka:

```scala
new ConfiguredEncoder[A]
      with DerivedInstance(
        constValue[mirror.MirroredLabel],
        summonLabels[mirror.MirroredElemLabels]
      ):

new ConfiguredDecoder[A]
      with DerivedInstance(
        constValue[mirror.MirroredLabel],
        summonLabels[mirror.MirroredElemLabels]
      ):
```

So `DerivedInstance` isn't adding any value.

Alternatively it could be used as a marker trait for encoders/decoders/codecs that were automatically derived. In that case it could be added to all the `Configured*` types but without any members.

Reply:
I see you deleted it, that works for me.